Core/HLE/sceNet.h

```diff
@@ -85,10 +147,39 @@ typedef struct ProductStruct { // Similar to SceNetAdhocctlAdhocId ?
 	s32_le unknown; // Unknown, set to 0 // Product Type ?
 	char product[PRODUCT_CODE_LENGTH]; // Game ID (Example: ULUS10000)
 } PACK ProductStruct;
+
 #ifdef _MSC_VER
 #pragma pack(pop)
 #endif
 
+typedef void(*sceNetApctlHandler)(int oldState, int newState, int event, int error, void* pArg);
+
+#define APCTL_PROFILENAME_MAXLEN 64
+#define APCTL_SSID_MAXLEN 32
+#define APCTL_IPADDR_MAXLEN 16
+#define APCTL_URL_MAXLEN 128
+typedef struct SceNetApctlInfoInternal { // Use struct instead of union for internal purpose
+	char 			name[APCTL_PROFILENAME_MAXLEN];
+	unsigned char 	bssid[ETHER_ADDR_LEN];
```

Comment:
I think you can safely declare these as char instead of unsigned char, so you then can use string functions without casting.

Reply:
Ah okay i'll change fields that contains null-terminated string to char, except that bssid since it's 6-byte of mac address and not null terminated one i should use memcpy instead